banyand/stream/stream_query_test.go

```diff
@@ -0,0 +1,532 @@
+	tester := assert.New(t)
+	s, deferFunc := setup(tester)
+	defer deferFunc()
+	baseTime := setupQueryData(tester, "multiple_shards.json", s)
+	tests := []struct {
+		name    string
+		args    queryOpts
+		want    shardsForTest
+		wantErr bool
+	}{
+		{
+			name: "all",
+			args: queryOpts{
+				entity:    tsdb.Entity{tsdb.AnyEntry, tsdb.AnyEntry, tsdb.AnyEntry},
+				timeRange: tsdb.NewTimeRangeDuration(baseTime, 1*time.Hour),
+			},
+			want: shardsForTest{
+				{
+					id:       0,
+					location: []string{"series_12243341348514563931", "data_flow_0"},
+					elements: []string{"1"},
+				},
+				{
+					id:       0,
+					location: []string{"series_1671844747554927007", "data_flow_0"},
+					elements: []string{"2"},
+				},
+				{
+					id:       1,
+					location: []string{"series_2374367181827824198", "data_flow_0"},
+					elements: []string{"5", "3"},
+				},
+				{
+					id:       1,
+					location: []string{"series_8429137420168685297", "data_flow_0"},
+					elements: []string{"4"},
+				},
+			},
+		},
+
+		{
+			name: "time range",
+			args: queryOpts{
+				entity:    tsdb.Entity{tsdb.AnyEntry, tsdb.AnyEntry, tsdb.AnyEntry},
+				timeRange: tsdb.NewTimeRangeDuration(baseTime.Add(1500*time.Millisecond), 1*time.Hour),
+			},
+			want: shardsForTest{
+				{
+					id:       0,
+					location: []string{"series_12243341348514563931", "data_flow_0"},
+				},
+				{
+					id:       0,
+					location: []string{"series_1671844747554927007", "data_flow_0"},
+				},
+				{
+					id:       1,
+					location: []string{"series_2374367181827824198", "data_flow_0"},
+					elements: []string{"5"},
+				},
+				{
+					id:       1,
+					location: []string{"series_8429137420168685297", "data_flow_0"},
+					elements: []string{"4"},
+				},
+			},
+		},
+		{
+			name: "find series by service_id and instance_id",
+			args: queryOpts{
+				entity:    tsdb.Entity{tsdb.Entry("webapp_id"), tsdb.Entry("10.0.0.1_id"), tsdb.AnyEntry},
+				timeRange: tsdb.NewTimeRangeDuration(baseTime, 1*time.Hour),
+			},
+			want: shardsForTest{
+				{
+					id:       0,
+					location: []string{"series_12243341348514563931", "data_flow_0"},
+					elements: []string{"1"},
+				},
+				{
+					id:       1,
+					location: []string{"series_2374367181827824198", "data_flow_0"},
+					elements: []string{"5", "3"},
+				},
+			},
+		},
+		{
+			name: "find a series",
+			args: queryOpts{
+				entity:    tsdb.Entity{tsdb.Entry("webapp_id"), tsdb.Entry("10.0.0.1_id"), convert.Uint64ToBytes(1)},
+				timeRange: tsdb.NewTimeRangeDuration(baseTime, 1*time.Hour),
+			},
+			want: shardsForTest{
+				{
+					id:       1,
+					location: []string{"series_2374367181827824198", "data_flow_0"},
+					elements: []string{"5", "3"},
+				},
+			},
+		},
+		{
+			name: "filter",
+			args: queryOpts{
+				entity:    tsdb.Entity{tsdb.AnyEntry, tsdb.AnyEntry, tsdb.AnyEntry},
+				timeRange: tsdb.NewTimeRangeDuration(baseTime, 1*time.Hour),
+				buildFn: func(builder tsdb.SeekerBuilder) {
```

Comment:
So the searchable tags (local indexes) can be directly processed by the storage layer now?

Reply:
Exactly.